frontend/server/src/test/java/org/pytorch/serve/ModelServerTest.java

```diff
@@ -1068,6 +1068,55 @@ public void testPredictionMemoryError() throws InterruptedException {
     @Test(
             alwaysRun = true,
             dependsOnMethods = {"testPredictionMemoryError"})
+    public void testSuccessBatch() throws InterruptedException {
+        int batch_size = 4
+        Channel channel = TestUtils.connect(ConnectorType.MANAGEMENT_CONNECTOR, configManager);
+        Assert.assertNotNull(channel);
+
+        TestUtils.setHttpStatus(null);
+        TestUtils.setResult(null);
+        TestUtils.setLatch(new CountDownLatch(1));
+
+        TestUtils.registerModel(channel, "noop.mar", "err_success", true, false, batch_size=batch_size);
+        TestUtils.getLatch().await();
+
+        StatusResponse status =
+                JsonUtils.GSON.fromJson(TestUtils.getResult(), StatusResponse.class);
+        Assert.assertEquals(
+                status.getStatus(),
+                "Model \"success_batch\" Version: 1.0 registered with 1 initial workers");
+
+        channel.close().sync();
+
+        channel = TestUtils.connect(ConnectorType.INFERENCE_CONNECTOR, configManager);
+        Assert.assertNotNull(channel);
+
+        TestUtils.setResult(null);
+        TestUtils.setLatch(new CountDownLatch(1));
+        TestUtils.setHttpStatus(null);
+        
+        for(int i = 0; i < batch_size ; i ++) {
+        DefaultFullHttpRequest req =
+            new DefaultFullHttpRequest(
+                   HttpVersion.HTTP_1_1, HttpMethod.POST, "/predictions/err_success");
+        // req.content().writeCharSequence("data=invalid_output", CharsetUtil.UTF_8);
+            HttpUtil.setContentLength(req, req.content().readableBytes());
+            req.headers()
+                .set(
+                        HttpHeaderNames.CONTENT_TYPE,
+                        HttpHeaderValues.APPLICATION_X_WWW_FORM_URLENCODED);
+        channel.writeAndFlush(req);
+
+        TestUtils.getLatch().await();
+        Assert.assertEquals(TestUtils.getHttpStatus(), HttpResponseStatus.ACCEPTED);
+        }
+        channel.close().sync();
+    }
+
+
+    @Test(
+            alwaysRun = true,
+            dependsOnMethods = {"testSuccessBatch"})
     public void testErrorBatch() throws InterruptedException {
```

Comment:
Why "testPredictionMemoryError" is replaced with "testSuccessBatch"?

Reply:
So the dependencies here just determine the order in which tests are ran, it's not a real dependency. All tests in this file are dependent on each other without needing it. We could perhaps have a a separate PR where we remove unnecessary dependencies but I believe this may be a known limitation because can't run more than one torchserve instance at a time so can't run more than one torchserve test at a time